course/categorize.php

```diff
@@ -96,16 +96,42 @@ function flattenarr($ar) {
 		$questionlibs[$row[0]][] = $row[1];
 		$libnames[$row[1]] = $row[2];
 	}
-	$query = "SELECT iq.id,iq.category,iqs.description FROM imas_questions AS iq,imas_questionset as iqs";
+
+	//add assessment names as options
+	//find the names of assessments these questionsetids appear in
+	$query = "SELECT imas_questions.id AS qid,imas_questions.questionsetid AS qsetid,imas_assessments.id AS aid,imas_assessments.name ";
+	$query .= "FROM imas_questions, imas_assessments ";
+	$query .= "WHERE imas_questions.assessmentid=imas_assessments.id ";
+	$query .= "AND imas_questions.questionsetid = ANY (SELECT imas_questions.questionsetid FROM imas_questions WHERE imas_questions.assessmentid='$aid') ";
+	$query .= "AND imas_assessments.courseid='$cid' ";
+	$query .= "ORDER BY aid";
+	$result = mysql_query($query) or die("Query failed : " . mysql_error());
+	$assessmentnames = array();
+	$qsetidassessment = array();
+	while ($row = mysql_fetch_array($result,MYSQL_ASSOC)) {
+		//store the relevent assessment names
+		$assessmentnames[$row['aid']] = $row['name'];
+		if ($row['aid']!=$aid) {
+			//remember this other assignment which uses this same questionsetid but don't add duplicate entries
+			if (!is_array($qsetidassessment[$row['qsetid']]) || !in_array($row['aid'],$qsetidassessment[$row['qsetid']])) {

```

Comment:
Yeah, it's fine.  Since you don't need the imas_questions.id, you _might_ be able to eliminate the need for it by changing the SELECT to:

`SELECT  DISTINCT imas_questions.questionsetid AS qsetid,imas_assessments.id AS aid, imas_assessments.name`

I think that would give you distinct qsetid/aid sets, but you'd want to test it out.